### PR TITLE
TransactionView: InstructionsMeta

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- Task in #2255

#### Summary of Changes
- Parse instructions

Fixes #

### CODE REVIEW

File: transaction-view/src/instructions_meta.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
+use crate::{
+    bytes::{advance_offset_for_array, check_remaining, optimized_read_compressed_u16, read_byte},
+    result::Result,
+};
+
+/// Contains meta-data about the instructions in a transaction packet.
```
Comment (LucasSte) [inline transaction-view/src/instructions_meta.rs]:
```suggestion
/// Contains metadata about the instructions in a transaction packet.
```

File: transaction-view/src/instructions_meta.rs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    use {
+        super::*,
+        solana_sdk::{instruction::CompiledInstruction, short_vec::ShortVec},
+    };
+
+    #[test]
+    fn test_zero_instructions() {
+        let bytes = bincode::serialize(&ShortVec(Vec::<CompiledInstruction>::new())).unwrap();
+        let mut offset = 0;
+        assert!(InstructionsMeta::try_new(&bytes, &mut offset).is_ok());
```
Comment (LucasSte) [inline transaction-view/src/instructions_meta.rs]: Should we check in this case that the number of instructions is zero?

File: transaction-view/src/instructions_meta.rs (hunk 1 above)
Comment (apfitzge) [reply transaction-view/src/instructions_meta.rs]: e8d1b22f4d2dacd7c181ff55c0d9e78c03286cd6

File: transaction-view/src/instructions_meta.rs (hunk 2 above)
Comment (apfitzge) [reply transaction-view/src/instructions_meta.rs]: Yeah idk how I missed that when writing.
I added some additional asserts on the offset as well: ed84c8000303512c07f673119306cdd9b515d3d4

**Review by LucasSte — APPROVED** [review]

**Pull request merged**